src/Http/Livewire/Settings/PriceLists.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+namespace FluxErp\Http\Livewire\Settings;
+
+use FluxErp\Actions\Discount\CreateDiscount;
+use FluxErp\Actions\Discount\UpdateDiscount;
+use FluxErp\Actions\PriceList\CreatePriceList;
+use FluxErp\Actions\PriceList\DeletePriceList;
+use FluxErp\Actions\PriceList\UpdatePriceList;
+use FluxErp\Models\Category;
+use FluxErp\Models\PriceList;
+use FluxErp\Models\Product;
+use Illuminate\Contracts\View\View;
+use Illuminate\Validation\ValidationException;
+use Livewire\Component;
+use WireUi\Traits\Actions;
+
+class PriceLists extends Component
+{
+    use Actions;
+
+    public array $selectedPriceList = [
+        'name' => '',
+        'parent_id' => null,
+        'price_list_code' => null,
+        'is_net' => false,
+        'is_default' => false,
+        'discount' => [
+            'discount' => null,
+            'is_percentage' => true,
+        ],
+    ];
+
+    public array $priceLists;
+
+    public array $discountedCategories = [];
+
+    public array $newCategoryDiscount = [
+        'category_id' => null,
+        'discount' => null,
+        'is_percentage' => true,
+    ];
+
+    public bool $editModal = false;
+
+    public function mount(): void
+    {
+        $this->priceLists = PriceList::query()
+            ->get()
+            ->toArray();
+    }
+
+    public function showEditModal(int $priceListId = null): void
+    {
+        $priceList = PriceList::query()
+            ->whereKey($priceListId)
+            ->with('discount')
+            ->first();
+
+        $this->selectedPriceList = $priceList?->toArray() ?: [
+            'name' => '',
+            'parent_id' => null,
+            'price_list_code' => null,
+            'is_net' => false,
+            'is_default' => false,
+            'discount' => [
+                'discount' => null,
+                'is_percentage' => true,
+            ],
+        ];
+
+        if (is_null($this->selectedPriceList['discount'])) {
+            $this->selectedPriceList['discount'] = [
+                'discount' => null,
+                'is_percentage' => true,
+            ];
+        }
+
+        if ($priceList) {
+            $this->discountedCategories = $priceList->discountedCategories()
+                ->where('model_type', Product::class)
+                ->orderBy('sort_number', 'DESC')
+                ->with([
+                    'discounts' =>
+                        fn ($query) => $query->where('category_price_list.price_list_id', $priceList->id)
+                            ->select(['id', 'discount', 'is_percentage']),
+                ])
+                ->get()
+                ->toArray();
+        }
+
+        $this->editModal = true;
+        $this->resetErrorBag();
```

Comment:
no need to reset this i guess as you are using the toast notifications to display validation errors